### PR TITLE
Fix typo in sylius_order state machine callback

### LINKED ISSUE #13688: [CoreBundle] WinzouStateMachine -> sylius_order.yml -> wrong wording
https://github.com/Sylius/Sylius/blob/1b0aa43ae8d2983a0c29675fbdb4398dafebdb56/src/Sylius/Bundle/CoreBundle/Resources/config/app/state_machine/sylius_order.yml#L95

Instead of `sylis_cancel_order` it shoud be `sylius_cancel_order`

### PR DESCRIPTION
**Pull request by vvasiloi**
| Q               | A
| --------------- | -----
| Branch?         | 1.11
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #13688, replaces #12762 and #12286
| License         | MIT

Credits to @DennisdeBest and @antiseptikk for the previous PRs and to @AymLaf for the reminder.

### CODE REVIEW

**Review by stloyd — APPROVED** [review]

**Review by GSadee — COMMENTED** [review]
It is a potential BC break IMO, so this comment is probably still valid: https://github.com/Sylius/Sylius/pull/10704#issuecomment-540441237 but wdyt @Zales0123 @lchrusciel?

**Comment by vvasiloi** [comment]
> It is a potential BC break IMO, so this comment is probably still valid: [#10704 (comment)](https://github.com/Sylius/Sylius/pull/10704#issuecomment-540441237) but wdyt @Zales0123 @lchrusciel?

This is not covered by the BC promise, but I can provide a compatibility layer nonetheless, so if it is overridden with the current name it will still work as expected, but also trigger a deprecation.

**Comment by vvasiloi** [comment]
Easier said than done... I didn't find a precedent for this, so I came up with this idea of writing a compiler pass that will alter the parameter that stores the state machine configuration as a BC layer. Let me know what you think.

**Review by loic425 — APPROVED** [review]

**Review by loic425 — DISMISSED** [review]
@vvasiloi A PHPUnit test for this compiler pass is missing.

**Comment by vvasiloi** [comment]
@GSadee I added the BC layer.
@loic425 I added the tests for the pass.

**Review by Roshyo — APPROVED** [review]

**Comment by Rafikooo** [comment]
Hi Folks! Yes, this typo hurts the eye, however, the problematic of the change due to the BC policy is too big compared to the benefit of no typo, so we decided to abandon this topic until the release of Sylius 2.0 💃

**Comment by vvasiloi** [comment]
I made it BC, even though it's not covered by the BC promise. What's wrong with it?

**Comment by Rafikooo** [comment]
After more reports from the community, we decided to reconsider this PR again :)

File: composer.json [hunk 1]
```diff
@@ -113,7 +114,7 @@
         "webmozart/assert": "^1.9",
         "willdurand/hateoas": "^3.0",
         "willdurand/hateoas-bundle": "^2.0",
-        "winzou/state-machine-bundle": "^0.5"
+        "winzou/state-machine-bundle": "~0.5"
```
Comment (GSadee) [inline composer.json:117]: Why?

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Victor and Rafał! :tada:

File: composer.json (hunk 1 above)
Comment (Rafikooo) [reply composer.json:117]: Necessary to unlock upgrades for the rest of the affected dependencies

**Pull request merged**